icu4c/source/test/intltest/usettest.cpp

```diff
@@ -4108,3 +4109,65 @@ void UnicodeSetTest::TestSkipToStrings() {
     assertNext(iter, u"ch");
     assertFalse("no next", iter.next());
 }
+
+void UnicodeSetTest::TestPatternCodePointComplement() {
+    IcuTestErrorCode errorCode(*this, "TestPatternCodePointComplement");
+    // ICU-21524 changes pattern ^ and equivalent functions to perform a "code point complement".
+    // [^abc{ch}] = [[:Any:]-[abc{ch}]] which removes all strings.
+    {
+        UnicodeSet simple(u"[^abc{ch}]", errorCode);
+        assertEquals("[^abc{ch}] --> lots of elements", 0x110000 - 3, simple.size());
```

Comment:
(completely optional) I'm a big fan of using the `WHERE` macro  in test assert messages. On test failures it leads you straight to the line in the test code, avoiding the need to search for the text of the message.

Reply:
I like test code to be as similar as possible in C++ vs. Java.